doc/api/net.md

```diff
@@ -246,168 +354,215 @@ server.on('error', (e) => {
-instances implement a duplex Stream interface.  They can be created by the
-user and used as a client (with [`connect()`][]) or they can be created by Node.js
-and passed to the user through the `'connection'` event of a server.
+The `net.Socket` object is an abstraction of a TCP or local socket. All
+`net.Socket` instances implement a [Duplex][] Stream interface.
+
+Instances may be created by users (e.g. using [`net.connect()`][]) or by
+Node.js and passed to the user through the `'connection'` event.
+
+All `net.Socket` objects are [`EventEmitter`][] instances.
 
 ### new net.Socket([options])
 <!-- YAML
 added: v0.3.4
 -->
 
-Construct a new socket object.
-
-`options` is an object with the following defaults:
-
-```js
-{
-  fd: null,
-  allowHalfOpen: false,
-  readable: false,
-  writable: false
-}
-```
-
-`fd` allows you to specify the existing file descriptor of socket.
-Set `readable` and/or `writable` to `true` to allow reads and/or writes on this
-socket (NOTE: Works only when `fd` is passed).
-About `allowHalfOpen`, refer to `createServer()` and `'end'` event.
+* `options` {Object}
+  * `fd` {number} Used to specify a numeric file descriptor of an existing
+    socket. Defaults to `null`
+  * `allowHalfOpen` {boolean} If `true`, allows the socket to be in a "half
+    open" state. Defaults to `false`
+  * `readable` {boolean} If `fd` is specified, setting `readable` to `true`
+    allows data from the socket to be read. Defaults to `false`
+  * `writable` {boolean} If `fd` is specified, setting `writabl` to `true`
+    allows data to be written to the socket. Defaults to `false`
 
-`net.Socket` instances are [`EventEmitter`][] with the following events:
+Construct a new `net.Socket` instance.
 
 ### Event: 'close'
 <!-- YAML
 added: v0.1.90
 -->
 
-* `had_error` {Boolean} `true` if the socket had a transmission error.
+The `'close'` event is emitted once the socket is fully closed.
 
-Emitted once the socket is fully closed. The argument `had_error` is a boolean
-which says if the socket was closed due to a transmission error.
+The listener callback is called with a single boolean `had_error` argument.
+If `true`, then the socket was closed due to a transmission error.
+
+```js
+socket.on('close', (had_error) => {
+  if (had_error)
+    console.log('There was a transmission error!');
+  else
+    console.log('The connection closed normally.');
+});
+```
 
 ### Event: 'connect'
 <!-- YAML
 added: v0.1.90
 -->
 
-Emitted when a socket connection is successfully established.
-See [`connect()`][].
+The `'connect'` event is emitted when a socket connection is successfully
+established. See [`net.connect()`][].
+
+The listener callback is called without passing any arguments.
+
+```js
+socket.on('connect', () => {
+  console.log('The socket is connected!');
+});
+```
 
 ### Event: 'data'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {Buffer}
+The `'data'` event is emitted when data is received.
```

Comment:
We might want to say that this is effectively how people are expected to read data off a socket since it might not be obvious to newbies - but that might better belong in a "working with sockets in Node" guide and not the docs.